### ISSUE TITLE
[Q] How do I direct W&B to stop a sweep with the Python API? (not with local controller)

### ISSUE THREAD

**Issue by fantauzzi** [wandb/wandb#5560]
Bayesian sweeps remain in a Running state indefinitely, unless stopped. I understand I could stop the sweep from the UI or from the command line (`wandb sweep --stop`). But how do I do it from a Python program, with the wandb SDK? Note I am not using a local controller.

**Comment by ArtsiomWB** [comment]
Hi @fantauzzi ! Could I see your sweep config and how you are setting up your sweep?

**Comment by fantauzzi** [comment]
This is the sweep configuration 
```
sweep_configuration = {
    'method': 'bayes',
        'name': 'sweep',
    'metric': {'goal': 'minimize', 'name': 'best_score_val'},
    'parameters':
        {
            'learning_rate': {'max': 0.3, 'min': 0.01},
            'depth': {'max': 10, 'min': 6},
            'border_count': {'value': 128}
        }
}
```
Not sure about the other question, anyway the code is short an uploaded here 
https://github.com/fantauzzi/taxi/blob/main/src/main.py

**Comment by fantauzzi** [comment]
As a workaround I can stop the sweep invoking the CLI from within Python, with something like
```
api = wandb.Api()
prj = api.project(params.wandb.project)
sweep_long_id = f'{prj.entity}/{params.wandb.project}/{sweep_id}'
command = f'wandb sweep --stop {sweep_long_id}'
system(command)
```
which is inconvenient. Would rather do it with the Python SDK, without spawning a subshell.

**Comment by ArtsiomWB** [comment]
Hi @fantauzzi! Thank you for sending over the examples. SDK-controlled sweeps is a long-standing feature request we keep on receiving. If you'd like to I can submit this feature request for you as well and bump up the priority on it. 

Otherwise, stopping the sweep your current way is the only way of doing it through code, unfortunately.

**Comment by fantauzzi** [comment]
Yes please, I would be grateful if you could submit the feature request.

**Comment by ArtsiomWB** [comment]
Perfect! your feature request has been submitted! I'll keep you posted if I'll happen to have any updates on it. Cheers!

**Comment by alegonz** [comment]
You can also stop a sweep using the internal API (which is what the CLI tool uses internally):
```python
import wandb.apis
wandb.apis.InternalApi().stop_sweep(entity="foo", project="bar", sweep="a1b2c3d4")
```
It is not ideal, but better than spawning a subprocess.
